File: php/features.php
<?php
declare(strict_types=1);

// LasVegasCity SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LasVegasCityFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LasVegasCityBaseFeature();
            case "test":
                return new LasVegasCityTestFeature();
            default:
                return new LasVegasCityBaseFeature();
        }
    }
}

File: py/lasvegascity_sdk/features.py
# LasVegasCity SDK feature factory

from lasvegascity_sdk.feature.base_feature import LasVegasCityBaseFeature
from lasvegascity_sdk.feature.test_feature import LasVegasCityTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LasVegasCityBaseFeature(),
        "test": lambda: LasVegasCityTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
